File: frontend/src/types/aiSummary.ts
export interface AiSummary {
  articleId: string
  highlights: string[]
  impactScope: string
  controversy: string
  editorView: string
  generatedAt: string
}

export interface AiReport {
  articleId: string
  tldr: string
  keyPoints: string[]
  pros: string[]
  cons: string[]
  timeline: AiReportTimelineItem[]
  scores: AiReportScores
  relatedTags: string[]
  editorNote: string
  rating: 'low-impact' | 'medium-impact' | 'high-impact' | 'watchlist'
  provider: string
  generatedAt: string
}

export interface AiReportTimelineItem {
  label: string
  description: string
}

export interface AiReportScores {
  impact: number
  confidence: number
  controversy: number
}

export interface AiReportStreamEvent {
  type: 'started' | 'status' | 'report' | 'completed' | 'error'
  message?: string
  code?: string
  report?: AiReport
}

File: backend/src/AiDaily.Domain/Entities/AiSummary.cs
namespace AiDaily.Domain.Entities;

public sealed class AiSummary
{
    public required string Id { get; init; }
    public required string ArticleId { get; init; }
    public IReadOnlyList<string> Highlights { get; init; } = Array.Empty<string>();
    public required string ImpactScope { get; init; }
    public required string Controversy { get; init; }
    public required string EditorView { get; init; }
    public required string Provider { get; init; }
    public required string PromptVersion { get; init; }
    public DateTimeOffset GeneratedAt { get; init; }
}
